lib/ansible/executor/task_executor.py

```diff
@@ -858,15 +855,14 @@ def _get_connection(self, variables, templar):
             display.vvvv('attempting to start connection', host=self._play_context.remote_addr)
             display.vvvv('using connection plugin %s' % connection.transport, host=self._play_context.remote_addr)
 
-            options = self._get_persistent_connection_options(connection, variables, templar)
+            options = self._get_persistent_connection_options(connection, cvars, templar)
             socket_path = start_connection(self._play_context, options, self._task._uuid)
             display.vvvv('local domain socket path is %s' % socket_path, host=self._play_context.remote_addr)
             setattr(connection, '_socket_path', socket_path)
 
         return connection
 
-    def _get_persistent_connection_options(self, connection, variables, templar):
-        final_vars = combine_vars(variables, variables.get('ansible_delegated_vars', dict()).get(self._task.delegate_to, dict()))
```

Comment:
This is a change so it could use a porting note, but looks like it's doing the right thing now.

Reply:
yep, missed this in first PR of series that fixes delegation vars .. this was still doing the old 'combining all vars' so you would get from inventory_hostname if the delegated host was not overriding.

tempted to say i should add porting note for the original PR and ref the whole list of them up to this one.